autoscheduler/frontend/src/tests/ReduxAvailability.test.tsx

```diff
@@ -0,0 +1,237 @@
+      store.dispatch(addAvailability(availability1));
+      store.dispatch(addAvailability(availability2));
+
+      // assert
+      expect(store.getState().availability).toEqual(expected);
+    });
+    test('with overlaps on both ends', () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer);
+      const availability1 = {
+        ...dummyArgs,
+        time1: makeTime(11, 30),
+        time2: makeTime(13, 42),
+      };
+      const availability2 = {
+        ...dummyArgs,
+        time1: makeTime(7, 0),
+        time2: makeTime(8, 30),
+      };
+      const availability3 = {
+        ...dummyArgs,
+        time1: makeTime(8, 0),
+        time2: makeTime(12, 0),
+      };
+      const expected = [{
+        ...dummyArgs,
+        startTimeHours: 7,
+        startTimeMinutes: 0,
+        endTimeHours: 13,
+        endTimeMinutes: 42,
+      }];
+
+      // act
+      store.dispatch(addAvailability(availability1));
+      store.dispatch(addAvailability(availability2));
+      store.dispatch(addAvailability(availability3));
+
+      // assert
+      expect(store.getState().availability).toEqual(expected);
+    });
+
+    test('only once when 3 or more coincide', () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer);
+      const availability1 = {
+        ...dummyArgs,
+        time1: makeTime(13, 0),
+        time2: makeTime(14, 0),
+      };
+      const availability2 = {
+        ...dummyArgs,
+        time1: makeTime(15, 0),
+        time2: makeTime(16, 0),
+      };
+      const availability3 = {
+        ...dummyArgs,
+        time1: makeTime(17, 0),
+        time2: makeTime(17, 30),
+      };
+      const availability4 = {
+        ...dummyArgs,
+        time1: makeTime(18, 0),
+        time2: makeTime(18, 30),
+      };
+      const availability5 = {
+        ...dummyArgs,
+        time1: makeTime(12, 0),
+        time2: makeTime(12, 0),
+      };
+      const availability5New = {
+        ...dummyArgs,
+        time1: makeTime(12, 0),
+        time2: makeTime(15, 30),
+      };
+      const expected: Availability[] = [
+        {
+          ...dummyArgs,
+          startTimeHours: 17,
+          startTimeMinutes: 0,
+          endTimeHours: 17,
+          endTimeMinutes: 30,
+        },
+        {
+          ...dummyArgs,
+          startTimeHours: 18,
+          startTimeMinutes: 0,
+          endTimeHours: 18,
+          endTimeMinutes: 30,
+        },
+        {
+          ...dummyArgs,
+          startTimeHours: 12,
+          startTimeMinutes: 0,
+          endTimeHours: 16,
+          endTimeMinutes: 0,
+        },
+      ];
+
+      // act
+      store.dispatch(addAvailability(availability1));
+      store.dispatch(addAvailability(availability2));
+      store.dispatch(addAvailability(availability3));
+      store.dispatch(addAvailability(availability4));
+      store.dispatch(addAvailability(availability5));
+      store.dispatch(updateAvailability(availability5New));
```

Comment:
What's the point of updating this availability? Ideally we wouldn't use other functions in our unit tests, as if `updateAvailability` is failing, this test will fail as well. Is this to get around not being able to mock mouse inputs?

Reply:
No, but it replicates how the UI would do it. When the user first clicks, it adds a 0-minute-long availability, then updates it as the user drags, and then merges when the user lets go.